Source/WebCore/layout/integration/inline/LayoutIntegrationLineLayout.cpp

```diff
@@ -988,7 +988,8 @@ bool LineLayout::hitTest(const HitTestRequest& request, HitTestResult& result, c
         if (!locationInContainer.intersects(boxRect))
             continue;
 
-        auto& elementRenderer = is<RenderElement>(renderer) ? downcast<RenderElement>(renderer) : *renderer.parent();
+        auto* renderElement = dynamicDowncast<RenderElement>(renderer);
+        auto& elementRenderer = renderElement ? *renderElement : *renderElement->parent();
```

Comment:
This is wrong. You want `renderer->parent()`, not `renderElement->parent()` since `renderElement` is null here.

Reply:
Could be written like:
```
auto* elementRenderer = dynamicDowncast<RenderElement>(renderer);
if (!elementRenderer)
    elementRenderer = renderer.parent;
```
or
```
auto& elementRenderer = *[&]() {
    auto* renderElement = dynamicDowncast<RenderElement>(renderer);
    return renderElement ? renderElement : renderElement->parent();
}();
```

Curious if @alanbaradlay has a preference here. Either way, the PR needs fixing.

Reply:
Doh, thanks for identifying where the mistake was!

Are lambdas essentially free then? Might also be a way to avoid putting variables in scope elsewhere.

Reply:
lambda setup is the preferred one.